sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -83,50 +83,71 @@ internal class NextLinkOperationImplementation : IOperation
 
         public static IOperation Create(
             HttpPipeline pipeline,
-            RehydrationToken? rehydrationToken,
-            string? apiVersionOverride = null)
+            RehydrationToken rehydrationToken,
+            bool skipApiVersionOverride = false,
+            string? apiVersionOverrideValue = null)
         {
             AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+            AssertNotNull(pipeline, nameof(pipeline));
 
             var lroDetails = ModelReaderWriter.Write(rehydrationToken!, ModelReaderWriterOptions.Json).ToObjectFromJson<Dictionary<string, string>>();
```

Comment:
Curious: why not get these values directly from the rehydration token?  It seems like a high performance cost to serialize to a dictionary to get values from a type where you could just read the values from properties.  Is it because it is internal shared source?  Would it make sense to make this type internal in Core instead of internal shared source so it could access the internal properties of `RehydrationToken`?

Reply:
As discussed in https://github.com/Azure/azure-sdk-for-net/pull/42686#discussion_r1538358875.

We want to keep the properties and constructor in `RehydrationToken` internal, so we can't access them in `NextLinkOperationImplementation` since it is internal shared source.
If we can take `NextLinkOperationImplementation` out of shared source, we can access internal directly. Make `NextLinkOperationImplementation` is not in the scope of this PR, I will try it out in a separate PR to see if I can afford the effort. Otherwise, we have to take this workaround. 

cc @m-nash  @KrzysztofCwalina @tg-msft

Reply:
`NextLinkOperationImplementation.Create` is used in all RPs, for instance [ResourceManager](https://github.com/Azure/azure-sdk-for-net/blob/db30db94d1acb4e9f475a4fe4d98d0680944ea34/sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/LongRunningOperation/ManagementGroupsArmOperation.cs#L34), [CosmosDB](https://github.com/Azure/azure-sdk-for-net/blob/db30db94d1acb4e9f475a4fe4d98d0680944ea34/sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/LongRunningOperation/CosmosDBArmOperation.cs#L34).
Added a TODO to access the internal members from `RehdyrationToken` directly once we remove `NextLinkOperationImplementation` from internal shared.
Another option is to make the internal members in `RehydrationToken` public.

Reply:
I agree that @KrzysztofCwalina is the best person to make the decision regarding whether serializing a public type to a dictionary is an acceptable performance cost to pay for this operation in the short term.  This allows us to work around one of the problems we have with shared source types.  If we decide to use this approach, we should add a comment to the code to come back and address it when we resolve https://github.com/Azure/azure-sdk-for-net/issues/28369

Reply:
Will handle this in https://github.com/Azure/azure-sdk-for-net/issues/43260